tests/httpwrappers/tests.py

```diff
@@ -470,6 +470,10 @@ def test_header_deletion(self):
         # del doesn't raise a KeyError on nonexistent headers.
         del r.headers['X-Foo']
 
+    def test_instantiate_with_headers(self):
+        r = HttpResponse('hello', headers={'X-Foo': 'foo'})
+        self.assertEqual(r.headers['X-Foo'], 'foo')
```

Comment:
Maybe test whether `r.headers['x-foo']` is set or something? (`CaseInsensitiveMapping` itself is probably thoroughly tested but it may be useful to verify the case insensitivity here too)

Reply:
I'm not exactly sure what it's protecting against, but seems reasonable, have done this.